v4.x backport - test: convert var->const/let in tests

gibfahn opened the pull request:
I'll fix up the commit messages once the tests are passing.

Backport of #10685, I couldn't cherry-pick as it brought in hundreds of unrelated changes, so I redid the `eslint --fix` and manual fixup for `v4.x`.

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [ ] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [ ] commit message follows commit guidelines

##### Affected core subsystem(s)
<!-- Provide affected core subsystem(s) (like doc, cluster, crypto, etc). -->
test

gibfahn commented:
I'm getting an error for tests that have strict mode disabled, e.g. [test-fs-stat#L40](https://github.com/nodejs/node/blob/master/test/parallel/test-fs-stat.js#L40). I resolved it by disabling the `no-var` lint rule for those lines. If anyone objects please let me know.

```
=== release test-fs-stat ===                                                   
Path: parallel/test-fs-stat
/Users/gib/wrk/com/node/test/parallel/test-fs-stat.js:40
  let stats;
  ^^^

SyntaxError: Block-scoped declarations (let, const, function, class) not yet supported outside strict mode
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:373:25)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Function.Module.runMain (module.js:441:10)
    at startup (node.js:140:18)
    at node.js:1043:3
Command: out/Release/node /Users/gib/wrk/com/node/test/parallel/test-fs-stat.js
```

Files failing:
 - test/parallel/test-fs-stat.js               
 - test/parallel/test-module-nodemodulepaths.js
 - test/parallel/test-net-binary.js            
 - test/parallel/test-repl.js                  
 - test/parallel/test-util-inspect.js

gibfahn commented:
CI: https://ci.nodejs.org/job/node-test-commit/8344/

I'd appreciate people taking a look at the first and the last two commits.

MylesBorins commented:
@gibfahn since v4.8.1 is likely going to be the last Active LTS release of v4.x and the first r.c. has already gone out it might make sense to skip this.

gibfahn commented:
@MylesBorins obviously I'd like you to merge it if possible (having just finished wrestling it into shape), but it's up to you. The benefit to having this is that virtually nothing will backport cleanly to `v4.x` without it (same is true to a lesser extent of #10698 as well). Should also be low-risk as it's just testcase changes.

However I guess if we're really not expecting almost anything to be added to `v4.x` in the future I guess it might not be worth it, ¯\\\_(ツ)\_/¯

gibfahn commented:
I've rebased and rerun CI anyway, so @MylesBorins feel free to whatever you think is best!

CI 2: https://ci.nodejs.org/job/node-test-commit/8360/

The branch from before I squashed is [var2const-v4.x-presquash](https://github.com/gibfahn/node/tree/var2const-v4.x-presquash), the `squash! manual` commits are the ones that might be worth reviewing.

MylesBorins commented:
I'm going to go ahead and close this for v4, will land the v6 version after this release
